Composer/packages/client/src/components/ProjectTree/ProjectTree.tsx

```diff
@@ -92,97 +112,270 @@ function sortDialog(dialogs: DialogInfo[]) {
-  dialogs: DialogInfo[],
-  dialogId: string,
-  filter: string
-): { items: any[]; groups: IGroup[] } {
-  let position = 0;
-  const result = dialogs
-    .filter((dialog) => {
-      return dialog.displayName.toLowerCase().includes(filter.toLowerCase());
-    })
-    .reduce(
-      (result: { items: any[]; groups: IGroup[] }, dialog) => {
-        const warningContent = containUnsupportedTriggers(dialog);
-        result.groups.push(createGroupItem(dialog, dialogId, position, warningContent));
-        position += dialog.triggers.length;
-        dialog.triggers.forEach((item, index) => {
-          const warningContent = triggerNotSupported(dialog, item);
-          result.items.push(createItem(item, index, warningContent));
-        });
-        return result;
-      },
-      { items: [], groups: [] }
-    );
-  return result;
-}
-
-interface IProjectTreeProps {
+type BotInProject = {
   dialogs: DialogInfo[];
-  dialogId: string;
-  selected: string;
-  onSelect: (id: string, selected?: string) => void;
+  projectId: string;
+  name: string;
+  isRemote: boolean;
+};
+
+type IProjectTreeProps = {
+  onSelect?: (link: TreeLink) => void;
+  onAllSelected?: () => void;
+  showTriggers?: boolean;
+  showDialogs?: boolean;
+  navLinks?: TreeLink[];
   onDeleteTrigger: (id: string, index: number) => void;
   onDeleteDialog: (id: string) => void;
-}
+};
 
-export const ProjectTree: React.FC<IProjectTreeProps> = (props) => {
-  const { onboardingAddCoachMarkRef } = useRecoilValue(dispatcherState);
+export const ProjectTree: React.FC<IProjectTreeProps> = ({
+  onAllSelected = undefined,
+  showTriggers = true,
+  showDialogs = true,
+  onDeleteDialog,
+  onDeleteTrigger,
+  onSelect,
+}) => {
+  const { onboardingAddCoachMarkRef, selectTo, navTo } = useRecoilValue(dispatcherState);
 
-  const groupRef: React.RefObject<IGroupedList> = useRef(null);
-  const { dialogs, dialogId, selected, onSelect, onDeleteTrigger, onDeleteDialog } = props;
   const [filter, setFilter] = useState('');
+  const [selectedLink, setSelectedLink] = useState<TreeLink | undefined>();
   const delayedSetFilter = debounce((newValue) => setFilter(newValue), 1000);
   const addMainDialogRef = useCallback((mainDialog) => onboardingAddCoachMarkRef({ mainDialog }), []);
+  const projectCollection = useRecoilValue<BotInProject[]>(botProjectSpaceSelector).map((bot) => ({
+    ...bot,
+    hasWarnings: false,
+  }));
+  const currentProjectId = useRecoilValue(currentProjectIdState);
 
-  const sortedDialogs = useMemo(() => {
-    return sortDialog(dialogs);
-  }, [dialogs]);
+  const notificationMap: { [projectId: string]: { [dialogId: string]: Diagnostic[] } } = {};
+  for (const bot of projectCollection) {
+    notificationMap[bot.projectId] = {};
+    // this seems to be the only way to get these notifications
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const dialogMessages = useRecoilValue(validateDialogSelectorFamily(bot.projectId));
```

Comment:
This goes against the hooks rules.  U can leverage the botprojectspace selector to get this value. You would just do

```
const botProjectspace = useRecoilValue(botProjectspaceselector) at the top.

Now on line 157 u can do this

const dialogmessage = botProjectspace.filter((projectId) => projecId === bot.projectId)
```

Reply:
Got it. Should be fixed now.